diff --git a/tests/lib/test-case-helper.c b/tests/lib/test-case-helper.c
index 0011b37..bd5c0ee 100644
--- a/tests/lib/test-case-helper.c
+++ b/tests/lib/test-case-helper.c
@@ -75,7 +75,7 @@ folks_test_case_add_test_helper (FolksTestCase *self,
 {
   FolksTestCaseWeakMethod *wm;
 
-  g_return_if_fail (self == (FolksTestCase *) test_target);
+  g_return_val_if_fail (self == (FolksTestCase *) test_target, NULL);
 
   /* This will never be freed, so make sure not to hold references. */
   wm = g_new0 (FolksTestCaseWeakMethod, 1);
