diff --git a/jhbuild/commands/tinderbox.py b/jhbuild/commands/tinderbox.py
index 672ae33..9023688 100644
--- a/jhbuild/commands/tinderbox.py
+++ b/jhbuild/commands/tinderbox.py
@@ -42,6 +42,9 @@ class cmd_tinderbox(BuildCommand):
             make_option('-c', '--clean',
                         action='store_true', dest='clean', default=False,
                         help=_('run make clean before make')),
+            make_option('--check',
+                        action='store_true', dest='check', default=False,
+                        help=_('run make check after building')),
             make_option('-n', '--no-network',
                         action='store_true', dest='nonetwork', default=False,
                         help=_('skip version control update')),
