Index: ChangeLog
===================================================================
--- ChangeLog	(revision 172586)
+++ ChangeLog	(working copy)
@@ -1,3 +1,16 @@
+2014-08-14  Ting-Wei Lan  <lantw44@gmail.com>
+
+        [GTK] Disable memory sampler on non-Linux system
+        https://bugs.webkit.org/show_bug.cgi?id=134483
+
+        Reviewed by NOBODY (OOPS!).
+
+        Source/WebKit2/Shared/linux/WebMemorySamplerLinux.cpp only works on
+        Linux because it uses many Linux-specific features, so we should disable
+        memory sampler on other systems by default.
+
+        * Source/cmake/OptionsGTK.cmake:
+
 2014-08-14  Carlos Garcia Campos  <cgarcia@igalia.com>
 
         [GTK] REGRESSION(r171598, CMake): Introspection files are installed at the wrong path
Index: Source/cmake/OptionsGTK.cmake
===================================================================
--- Source/cmake/OptionsGTK.cmake	(revision 172586)
+++ Source/cmake/OptionsGTK.cmake	(working copy)
@@ -96,6 +96,12 @@ else ()
     set(WebKit2_VERSION_SCRIPT "-Wl,--version-script,${CMAKE_MODULE_PATH}/gtksymbols.filter")
 endif ()
 
+if (CMAKE_SYSTEM_NAME MATCHES "Linux")
+    WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_MEMORY_SAMPLER ON)
+else ()
+    WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_MEMORY_SAMPLER OFF)
+endif ()
+
 # FIXME: We want to expose fewer options to downstream, but for now everything is public.
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_3D_RENDERING ON)
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_ACCESSIBILITY ON)
@@ -127,7 +133,6 @@ WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_LINK_PREFETCH OFF)
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_MEDIA_CAPTURE OFF)
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_MEDIA_CONTROLS_SCRIPT ON)
-WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_MEMORY_SAMPLER ON)
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_MHTML ON)
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_NAVIGATOR_CONTENT_UTILS OFF)
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_NAVIGATOR_HWCONCURRENCY ON)
